server/src/main/java/io/druid/server/coordination/ServerType.java

```diff
@@ -19,6 +19,18 @@
 
 package io.druid.server.coordination;
 
+/**
+ * This enum represents types of druid services that hold segments.
+ *
+ * These types are externally visible (e.g., from the output of /druid/coordinator/v1/servers).
+ *
+ * For backwards compatibility, when presenting these types externally, the toString() representation
+ * of the enum should be used.
+ *
+ * The toString() method converts the enum name() to lowercase and replaces underscores with hyphens,
+ * which is the format expected for the server type string prior to the patch that introduced ServerType:
+ * https://github.com/druid-io/druid/pull/4148
+ */
 public enum ServerType
```

Comment:
I don't think there are instances where a ServerType is being deserialized directly, DruidServer and DruidServerMetadata would be such locations but they currently take a String "type" and DruidServerMetadata converts that String to a ServerType with fromString().

I'll look into adding @JsonCreator here and changing DruidServer/DruidServerMetadata to accept a ServerType parameter.